docs/src/examples/components/RadioGroup/Types/RadioGroupExample.shorthand.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { Divider, RadioGroup } from '@stardust-ui/react'
+
+const items = [
+  { name: 'pizza', key: 'Capricciosa', label: 'Capricciosa', value: 'capricciosa' },
+  { name: 'pizza', key: 'Prosciutto', label: 'Prosciutto', value: 'prosciutto', disabled: true },
+  {
+    name: 'pizza',
+    key: 'Margherita',
+    label: 'Margherita',
+    value: 'margherita',
+  },
+]
+
+class RadioGroupExample extends React.Component {
+  state = { selectedValue: '' }
+  handleChange = (e, props) => {
+    this.setState({ selectedValue: props.value })
+  }
+  render() {
+    const { selectedValue } = this.state
+    return (
+      <div>
+        The selected value is: {selectedValue}
+        <Divider />
+        <RadioGroup
+          defaultCheckedValue="capricciosa"
+          items={items}
+          checkedValueChanged={this.handleChange}
+        />
+      </div>
+    )
+  }
+}
```

Comment:
I'm confused why the exactly same result, degining the RadioGroup.items as a components is not working. I am testing with this example and expect to achieve the same: 
```jsx
import React from 'react'
import { Divider, RadioGroup } from '@stardust-ui/react'
const props = [
  { name: 'pizza', key: 'Capricciosa', label: 'Capricciosa', value: 'capricciosa' },
  { name: 'pizza', key: 'Prosciutto', label: 'Prosciutto', value: 'prosciutto', disabled: true },
  {
    name: 'pizza',
    key: 'Margherita',
    label: 'Margherita',
    value: 'margherita',
  },
]
const items = [
  <RadioGroup.Item {...props[0]} />,
  <RadioGroup.Item {...props[1]} />,
  <RadioGroup.Item {...props[2]} />
]

class RadioGroupExample extends React.Component {
  state = { selectedValue: '' }
  handleChange = (e, props) => {
    this.setState({ selectedValue: props.value })
  }
  render() {
    const { selectedValue } = this.state
    return (
      <div>
        The selected value is: {selectedValue}
        <Divider />
        <RadioGroup
          defaultCheckedValue="capricciosa"
          items={items}
          checkedValueChanged={this.handleChange}
        />
      </div>
    )
  }
}
export default RadioGroupExample

```

Reply:
I'm not sure if passing components in items array is supposed to work. If you would pass the components as children, it would partially work. However similarly to Menu, RadioGroup works correctly only when you use shorthand for passing the items.

Reply:
This is still shorthand, it is just another way how the component can be passed. We can take a look together into this. I wasn't able to figure out yesterday what was the problem, but I am assuming is the casting of the item as IRadioGroupItemProps.